broadcast/util/routes.py

```diff
@@ -381,10 +385,17 @@ def get_role_needs_login_message(self):
     def get_role_method_whitelist(self):
         return self.role_method_whitelist
 
+    def get_role_xhr_method_whitelist(self):
+        return self.role_xhr_method_whitelist
+
     def check_role(self):
         if self.request.method in self.get_role_method_whitelist():
             return
 
+        if (self.request.is_xhr and
```

Comment:
Maybe `role_method_whitelist` and `role_xhr_method_whitelist` should be checked separately? I.e., if is_xhr, then check the xhr list, otherwise check the normal method whitelist.